x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -715,31 +742,66 @@ class AgentPolicyService {
       return;
     }
 
-    const policy = await agentPolicyService.get(soClient, agentPolicyId);
-    const fullPolicy = await agentPolicyService.getFullAgentPolicy(soClient, agentPolicyId);
-    if (!policy || !fullPolicy || !fullPolicy.revision) {
-      return;
-    }
+    const policies = await agentPolicyService.getByIDsAsMap(soClient, agentPolicyIds);
+    const fullPolicies = await Promise.all(
+      agentPolicyIds.map((agentPolicyId) =>
+        // TODO: make this a bulk
```

Comment:
making this a bulk seemed a bit more involved so wanted to get some thoughts on logic so far and whether this would even be needed.

Reply:
Taking a bit deeper look at `getFullAgentPolicy`, using it this way will fetch a few objects that will be re-used across agent policies (outputs, settings, upgrade download source uri), so we could save some round trips to ES in doing these fetches once and re-using them here.

This will likely only be a problem for users that have many thousands of agent policies, which we do have some customers who use that many. 

That said, this migration is a one time event for users upgrading from <8.5 so I don't want to spend a whole of time refactoring this if we can avoid it. @kpollich what do you think about deferring on optimizing this?

Reply:
I'm +1 on deferring those optimizations here with the caveat that we capture essentially what @joshdover has said above in a code comment replacing the current `TODO`